types/lodash/fp/index.d.ts

```diff
@@ -0,0 +1,618 @@
+         * value of the given element.
+         *
+         * If a value is also provided for thisArg the created _.matchesProperty style callback returns true for
+         * elements that have a matching property value, else false.
+         *
+         * If an object is provided for iteratee the created _.matches style callback returns true for elements that
+         * have the properties of the given object, else false.
+         *
+         * @param term
+         * @param collection The collection to iterate over.
+         * @return Returns the composed aggregate object.
+         */
+        keyBy<T>(termn: string): (collection: List<T>) => T;
+        keyBy<T>(termn: string, collection: List<T>): T;
+
+        /**
+         * Gets the first element of array.
+         *
+         * @alias _.first
+         *
+         * @param array The array to query.
+         * @return Returns the first element of array.
+         */
+        head<T>(array: List<T>): T;
+
+        flatten();
+
+        /**
+         * Iterates over elements of collection, returning the first element predicate returns truthy for.
+         * The predicate is bound to thisArg and invoked with three arguments: (value, index|key, collection).
+         *
+         * If a property name is provided for predicate the created _.property style callback returns the property
+         * value of the given element.
+         *
+         * If a value is also provided for thisArg the created _.matchesProperty style callback returns true for
+         * elements that have a matching property value, else false.
+         *
+         * If an object is provided for predicate the created _.matches style callback returns true for elements that
+         * have the properties of the given object, else false.
+         *
+         * @param predicate The function invoked per iteration.
+         * @param collection The collection to search.
+         * @return Returns the matched element, else undefined.
+         */
+        find<T>(predicate: ListIterator<T, boolean>, collection: List<T>): T;
+        find<T>(predicate: ListIterator<T, boolean>): (collection: List<T>) => T;
+
+        /**
+         * @see fp.find
+         */
+        find<TObject extends {}, T>(predicate: TObject, collection: List<T>): T;
+        find<TObject extends {}, T>(predicate: TObject): (collection: List<T>) => T;
+
+        /**
+         * Creates a function that returns the property value at path on a given object.
+         *
+         * @param path The path of the property to get.
+         * @return Returns the new function.
+         */
+        property<TObj, TResult>(path: string): (obj: TObj) => TResult;
+
+        /**
+         * Creates an array of the own enumerable property values of object.
+         *
+         * @param object The object to query.
+         * @return Returns an array of property values.
+         */
+        values<T>(object?: any): T[];
```

Comment:
This violates the common mistakes - you really just have a disguised type assertion. Should probably be `values(object?: any): any[];`

Reply:
This doesn't make sense to be optional either.